raisinbread/test/api/LorisApiVisitsTest.php

```diff
@@ -114,7 +114,162 @@ public function testGetCandidatesCandidVisit(): void
      */
     public function testPutCandidatesCandidVisit(): void
     {
-        $this->markTestSkipped('No access to create visits for any site');
+        // Test changing the Project & Battery
+        $json     = ['CandID'  => '900000',
+            'Visit'   => "V1",
+            'Site'    => "Data Coordinating Center",
+            'Battery' => "High Yeast",
+            'Project' => "Rye",
+        ];
+        $response = $this->client->request(
+            'PUT',
+            "candidates/900000/V1",
+            [
+                'headers' => $this->headers,
+                'json'    => $json
+            ]
+        );
+        // Verify the status code
+        $this->assertEquals(204, $response->getStatusCode());
+        // Verify the endpoint has a body
+        $body = $response->getBody();
+        $this->assertNotEmpty($body);
+
+        /**
+        * Test changing from a site with no affiliation to a site with affiliation
+        * Candidate 400266 is from site Rome. The test user only has access to
+        * Data Coordinating Center. He should not be able to modify a visit 
+        * for a candidate from a site he has no access to.
+        */
+        $json     = ['CandID'  => '400266',
+            'Visit'   => "V3",
+            'Site'    => "Data Coordinating Center",
+            'Battery' => "Stale",
+            'Project' => "Pumpernickel",
+        ];
+        $response = $this->client->request(
+            'PUT',
+            "candidates/400266/V3",
+            [
+                'headers' => $this->headers,
+                'json'    => $json,
+                'http_errors' => false
+            ]
+        );
+        // Verify the status code
+        $this->assertEquals(403, $response->getStatusCode());
+        // Verify the endpoint has a body
+        $body = $response->getBody();
+        $this->assertNotEmpty($body);
+
+        $json     = ['CandID'  => "115788",
+            'Visit'   => "V3",
+            'Site'    => "Data Coordinating Center",
+            'Battery' => "Stale",
+            'Project' => "Pumpernickel",
+        ];
+        $response = $this->client->request(
+            'PUT',
+            "candidates/115788/V3",
+            [
+                'headers'     => $this->headers,
+                'json'        => $json,
+                'http_errors' => false
+            ]
+        );
+        // verify the status code
+        $this->assertequals(409, $response->getstatuscode());
+        // verify the endpoint has a body
+        $body = $response->getbody();
+        $this->assertnotempty($body);
+
+        // Test assigning site with no affiliation
```

Comment:
What is the difference between this test and the second one? Could you explain in a comment?

Reply:
I tried to give more details in the comment of this test. The second test asserts that a visit can't be modified from a site without affiliation to a site with affiliation (Rome -> DCC). This one asserts that a visit can't be modified from a site without affiliation to a site with affiliation to a site without affiliation (DCC -> Montreal).